tests/cli/CommandLineParserTest.php

```diff
@@ -0,0 +1,213 @@
+		$result = $this->runCommand('optionalAndRequiredOptions', '');
+
+		self::assertEquals(['required' => 'invalid input: required cannot be empty'], $result->errors);
+	}
+
+	public function testOptionSetWithDeprecatedAliasGeneratesDeprecationWarningAndReturnsValue(): void {
+		$result = $this->runCommandReadingStandardError('optionalOptions', '--deprecated-string=string');
+
+		self::assertEquals('FreshRSS deprecation warning: the CLI option(s): deprecated-string are deprecated ' .
+				'and will be removed in a future release. Use: string instead',
+			$result
+		);
+
+		$result = $this->runCommand('optionalOptions', '--deprecated-string=string');
+
+		self::assertEquals('string', $result->string);
+	}
+
+	public function testAlwaysReturnUsageMessageWithUsageInfoForAllOptions(): void {
+		$result = $this->runCommand('optionalAndRequiredOptions', '');
+
+		self::assertEquals('Usage: cli-parser-test.php --required=<required> [-s --string=<string>] [-i --int=<int>] [-b --bool=<bool>] [-f --flag]',
+			$result->usage,
+		);
+	}
+
+	public static function optionalOptions(): stdClass {
+		$parser = new CommandLineParser();
+		$parser->addOption('string', (new Option('string', 's'))->deprecatedAs('deprecated-string'));
+		$parser->addOption('int', (new Option('int', 'i'))->typeOfInt());
+		$parser->addOption('bool', (new Option('bool', 'b'))->typeOfBool());
+		$parser->addOption('arrayOfString', (new Option('array-of-string', 'a'))->typeOfArrayOfString());
+		$parser->addOption('defaultInput', (new Option('default-input', 'i')), 'default');
+		$parser->addOption('optionalValue', (new Option('optional-value', 'o'))->withValueOptional());
+		$parser->addOption('optionalValueWithDefault', (new Option('optional-value-with-default', 'd'))->withValueOptional('true')->typeOfBool());
+		$parser->addOption('defaultInputAndOptionalValueWithDefault',
+			(new Option('default-input-and-optional-value-with-default', 'e'))->withValueOptional('optional'),
+			'default'
+		);
+		$parser->addOption('flag', (new Option('flag', 'f'))->withValueNone());
+
+		return $parser->parse(stdClass::class);
+	}
+
+	public static function optionalAndRequiredOptions(): stdClass {
+		$parser = new CommandLineParser();
+		$parser->addRequiredOption('required', new Option('required'));
+		$parser->addOption('string', new Option('string', 's'));
+		$parser->addOption('int', (new Option('int', 'i'))->typeOfInt());
+		$parser->addOption('bool', (new Option('bool', 'b'))->typeOfBool());
+		$parser->addOption('flag', (new Option('flag', 'f'))->withValueNone());
+
+		return $parser->parse(stdClass::class);
+	}
+
+	private function runCommand(string $static_method, string $options = ''): stdClass {
+		$command = __DIR__ . '/cli-parser-test.php';
+
+		$result = shell_exec("CLI_PARSER_TEST_STATIC_METHOD='$static_method' $command $options 2>/dev/null");
+		$result = is_string($result) ? unserialize($result) : '';
+
+		/** @var stdClass $result */
```

Comment:
There is probably a need of an explicit PHP cast and not only an annotation (this might explain the PHPStan failures with PHP 8.1).

Reply:
This has been dealt with in [25a673a](https://github.com/FreshRSS/FreshRSS/pull/6099/commits/25a673a8f7429deb1e8d50a596eb24d1bb07c48b) as part of the move away from dynamic properties.

Reply:
This looks quite good, thanks 👍🏻